tests/base.py

```diff
@@ -113,21 +122,36 @@ def expected_metadata(self):
         }
 
     def expected_streams(self):
+        """
+        Returns expected streams for tap. Intentionally removed the stream "stories"
+        as the records for this get update at a very high pace resulting it to consume
+        infinite time for syncing.
+        """
         return set(self.expected_metadata().keys()) - {'stories'}
 
     def expected_replication_keys(self):
+        """
+        Returns expected replication keys for streams in tap.
+        """
         return {table: properties.get(self.REPLICATION_KEYS, set()) for table, properties
                 in self.expected_metadata().items()}
 
     def expected_primary_keys(self):
+        """
+        Returns expected primary keys for streams in tap.
+        """
         return {table: properties.get(self.PRIMARY_KEYS, set()) for table, properties
                 in self.expected_metadata().items()}
 
     def expected_replication_method(self):
+        """
+        Returns expected replication method for streams in tap.
+        """
         return {table: properties.get(self.REPLICATION_METHOD, set()) for table, properties
                 in self.expected_metadata().items()}
 
-    def select_found_catalogs(self, conn_id, catalogs, only_streams=None, deselect_all_fields: bool = False, non_selected_props=[]):
+    def select_found_catalogs(self, conn_id, catalogs, only_streams=None,
+                              deselect_all_fields: bool = False, non_selected_props=[]):
```

Comment:
I would prefer: not defining non_selected_props=[] (by default) in the function argument because on line 161 you're eventually defining it to empty list in else block

Reply:
In below `non_selected_properties` is defined as a default argument which  is different than `non_selected_properties` on L#161.

`non_selected_properties = non_selected_props if not deselect_all_fields else []`